### ISSUE TITLE
Date format hard to parse for applications

### ISSUE THREAD

**Issue by glorialiou** [seattle-uat/civiform#1905]
The current date format used for applications is hard to parse (see here: https://docs.google.com/presentation/d/1goxiepPXPrW3wK8wUSNoCpcf7iXIohXfj96iTYiQxo0/edit#slide=id.gfd2a1448fa_0_1).

Request to change to something easier to read, e.g. Feb 10, 2022.
May need discussion about whether or not to add hours/mins and if so, whether to use local time or something else.
I suspect it may not be necessary to include hours/mins since admins are likely trying to gage whether an application was turned in by a certain date, but could be good to double check.

**Comment by rahaghas** [comment]
Change this to (for instance) Last updated: Jan 12, 2022 - using the first 3 letters of the month for each month.

**Comment by clouser2** [comment]
The UI referenced in the slide is the "Applications" UI here:

<img width="442" alt="Screen Shot 2022-05-24 at 10 22 36 AM" src="https://user-images.githubusercontent.com/1870301/170095405-cde61130-1c43-42ad-954a-dc2c0ffdb229.png">

Note: The image is for staging data that contains no real data.

For now, I'm going to keep the time on here as well, unless we think it's not adding value. We render dates in other places throughout the UI, including the "last edited timestamp" for programs. For consistency, I'm going to switch the date format for that as well (unless anyone objects).

**Comment by clouser2** [comment]
Estimated effort: <1 hour.

**Comment by clouser2** [comment]
@rahaghas I added you as a reviewer on #2556 to confirm this is acceptable to you.
